.github/scripts/generate_ci_workflows.py

```diff
@@ -83,11 +92,39 @@ class OperatingSystem:
     MACOS = "macos"
     MACOS_ARM64 = "macos-arm64"
 
+def import_module(modname: str, fname: str) -> ModuleType:
+    import importlib.util
+    spec = importlib.util.spec_from_file_location(modname, fname)
+    if spec is None:
+        raise ImportError(f"Could not load spec for module '{modname}' at: {fname}. \
+        Please run the regenerate.sh from .github folder before running generate_ci_workflows")
+    module = importlib.util.module_from_spec(spec)
+    try:
+        assert spec.loader is not None
+        assert module is not None
+        sys.modules[modname] = module
```

Comment:
Why does it needs to be registered globally?

Reply:
I am getting following error without registration:
```
Traceback (most recent call last):
  File "generate_ci_workflows.py", line 130, in <module>
    generate_binary_build_matrix = import_module(
  File "generate_ci_workflows.py", line 124, in import_module
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File ".tools/generate_binary_build_matrix.py", line 21, in <module>
    mod = sys.modules[__name__]
KeyError: 'generate_binary_build_matrix'
```
Python version : Python 3.8.6
This conforms with the doc: https://docs.python.org/3.8/library/importlib.html#importing-a-source-file-directly